openvdb/openvdb/tools/Diagnostics.h

```diff
@@ -441,7 +442,7 @@ struct CheckNormGrad
         : acc(grid.getConstAccessor())
         , invdx2(ValueType(1.0/math::Pow2(grid.voxelSize()[0])))
         , minVal2(_min*_min)
-        , maxVal2(_max*_max)
+        , maxVal2(_max < FLT_MAX ? _max*_max : _max)
```

Comment:
Although this tool asserts that it can only be used with floating-point, it can be used with other floating-point types, such as double or potentially half, so we can't use FLT_MAX here.

I would also prefer that we slightly re-engineer this to use `std::numeric_limits<ValueType>::max()` instead of using the C-style macro.